### PR TITLE
Media type renderers

### PR DESCRIPTION
**Pull request by MikeSperone**
## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

As per issue #3231, this allows the publisher to define a renderer to be implemented only in a specific mediaType.

A publisher could use it like this:
```
pbjs.addAdUnit({
    code: 'video1',
      // This renderer would work the same as it does now...
    renderer: {
        url: 'example.com/publishersCustomRenderer.js',
        render: function(bid) { renderAdUnit(...) }
    },
    mediaTypes: {
        video: {
            context: 'outstream',
            playerSize: [640, 480],
              // but a renderer passed in here would apply only to this mediaType.
              // It would override the above renderer if that was defined. ...
            renderer: {
                url: 'example.com/videoRenderer.js',
                render: function (bid) { renderVideo(...) }
            }
        },
        display: {
            ...,
              // and you'd be able to do the same for each mediaType
            renderer: {
                url: 'example.com/displayRenderer.js',
                render: function(bid) { renderDisplay(...) }
            }
        },
    },
    ...
});
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.
OK

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:
https://github.com/prebid/prebid.github.io/pull/2343

## Other information

Discussed in issue #3231

### CODE REVIEW

**Comment by MikeSperone** [comment]
it looks like this failed due to the zeotap test, which was fixed in the next build... I don't think I can re-run the build

**Comment by patmmccann** [comment]
This is really really helpful, thank you! @mike-chowla or @ncolletti (reviewers on a recent related pr) could you take a look as well?

File: package-lock.json [hunk 1]
```diff
@@ -7796,13 +7796,22 @@
         "function-bind": "^1.1.1",
         "has": "^1.0.3",
         "has-symbols": "^1.0.1",
-        "is-callable": "^1.2.0",
-        "is-regex": "^1.1.0",
+        "is-callable": "^1.1.5",
```
Comment (patmmccann) [inline package-lock.json:7799]: are these changes to package lock intended

Comment (MikeSperone) [reply package-lock.json:7799]: no, I didn't do anything intentional to change this

**Review by robertrmartinez — APPROVED** [review]
Looks good to me.

Second review needed

**Comment by robertrmartinez** [comment]
Added @ncolletti as second review since @patmmccann mentioned he reviewed a related PR.

**Comment by ncolletti** [comment]
Nice work! Looks good to me.. LGTM

**Review by ncolletti — APPROVED** [review]

**Pull request merged**